src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -3952,7 +3952,7 @@ void emitter::emitIns_I(instruction ins, emitAttr attr, cnsval_ssize_t val)
 {
     UNATIVE_OFFSET sz;
     instrDesc*     id;
-    bool           valInByte = ((signed char)val == val);
+    bool           valInByte = ((signed char)val == (target_ssize_t)val);
```

Comment:
are the higher 32 bits initialize when we cross-compile for x86 on x64?

Reply:
Unfortunately, the answer here is they may or may not be initialized.

If the constant value is a pointer to something like a methoddesc or whatnot, then the full 64bits are initialized, but if its a regular constant, only the lower 32bits are meaningful.

Switching to use cnssval_ssize_t to fix the problem isn't actually helpful, because on X86, cnssval_ssize_t IS ssize_t. I recently added cnssval_ssize_t in order to make the x86 jit work cross-architecture, but as per the comment on the typedef its a somewhat confusing type. I agree though that it might be reasonable to make the functions you call out as taking cnsval_ssize_t as parameter.